### PR TITLE
Introduce ALB Virtual Service management functions

### PR DESCRIPTION
**Pull request by Didainius**
**Note** Only to be reviewed once #414 is merged (because it contains and relies on that code). 

This PR adds methods and types to deal with NSX-T ALB Virtual Services.

### CODE REVIEW

File: .changes/v2.14.0/414-features.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+* Added support for ALB Pool to NSX-T Edge Gateway via type `NsxtAlbPool` and functions `GetAllAlbPools`,
+  `GetAllAlbPoolSummaries`, `GetAlbPoolByName`, `GetAlbPoolById`, `CreateNsxtAlbPool`, `Update`, `Delete`  [GH-414]
```
Comment (lvirbalas) [inline .changes/v2.14.0/414-features.md:2]: Nit:
```suggestion
  `GetAllAlbPoolSummaries`, `GetAlbPoolByName`, `GetAlbPoolById`, `CreateNsxtAlbPool`, `Update`, `Delete` [GH-414]
```

File: .changes/v2.14.0/417-features.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+* Added support for ALB Virtual Services to NSX-T Edge Gateway via type `NsxtAlbVirtualService` and functions `GetAllAlbVirtualServices`,
+  `GetAllAlbGetAllAlbVirtualServiceSummaries`, `GetAlbVirtualServiceByName`, `GetAlbVirtualServiceById`,
+  `CreateNsxtAlbVirtualService`, `NsxtAlbVirtualService.Update`, `NsxtAlbVirtualService.Delete`  [GH-417]
```
Comment (lvirbalas) [inline .changes/v2.14.0/417-features.md:3]: Nit:
```suggestion
  `CreateNsxtAlbVirtualService`, `NsxtAlbVirtualService.Update`, `NsxtAlbVirtualService.Delete` [GH-417]
```

**Review by lvirbalas — APPROVED** [review]
Great to see so many tests!
Just couple of total nits in this review.

File: govcd/nsxt_alb_service_engine_group_assignment.go [hunk 3]
```diff
@@ -108,6 +100,31 @@ func (vcdClient *VCDClient) GetAlbServiceEngineGroupAssignmentByName(name string
...
+	allServiceEngineGroupAssignments, err := vcdClient.GetAllAlbServiceEngineGroupAssignments(queryParameters)
+	if err != nil {
+		return nil, err
+	}
+
+	var foundGroup *NsxtAlbServiceEngineGroupAssignment
+
+	for _, serviceEngineGroupAssignment := range allServiceEngineGroupAssignments {
+		if serviceEngineGroupAssignment.NsxtAlbServiceEngineGroupAssignment.ServiceEngineGroupRef.Name == name {
+			foundGroup = serviceEngineGroupAssignment
```
Comment (vbauzysvmware) [inline govcd/nsxt_alb_service_engine_group_assignment.go:117]: Is the name unique? in case we don't return only first one found

**Review by vbauzysvmware — APPROVED** [review]

File: govcd/nsxt_alb_service_engine_group_assignment.go (hunk 3 above)
Comment (Didainius) [reply govcd/nsxt_alb_service_engine_group_assignment.go:117]: It must be unique:
image

**Review by dataclouder — APPROVED** [review]

**Pull request merged**